### ISSUE TITLE
Error: Certificate signed by unknown authority when creating a cluster

### ISSUE THREAD

**Issue by qasimalbaqali** [pulumi/pulumi-mongodbatlas#47]
Creating a cluster using the JavaScript library throws an error. Both the public and private key are set using the `mongodb.config` attributes.

Setting the public and private key using the config attributes, as environment variables or as pulumi config all give back the same error.

```javascript
mongodb.config.publicKey = "XXX";
mongodb.config.privateKey = "XX-XX-XX-XX-XX";

var mongoCluster = new mongodb.Cluster("rawag-dev", {
  autoScalingDiskGbEnabled: false,
  diskSizeGb: 100,
  mongoDbMajorVersion: "4.2",
  numShards: 1,
  projectId: "5f6fa64dfa46d90c51def6fd",
  providerBackupEnabled: true,
  providerDiskIops: 300,
  providerEncryptEbsVolume: true,
  providerInstanceSizeName: "M2",
  providerName: "AWS",
  providerRegionName: "EU_CENTRAL_1",
  providerVolumeType: "STANDARD",
  replicationFactor: 1,
});
```

```
error: error creating MongoDB Cluster: Post "https://cloud.mongodb.com/api/atlas/v1.0/groups/5f6fa64dfa46d90c51def6fd/clusters": x509: certificate signed by unknown authority
```

I am guessing that I need to enable the "Self-managed X.509 Authentication" option on MongoDB and provide it a certificate. Does pulumi do it automatically, or is there a pre-step that needs to be done manually? 

Is there an alternative when authenticating with MongoDB Atlas? Does creating a Database User and connecting it to an IAM role/user is something that is valid for authentication?

**Comment by stack72** [comment]
Hi @qasimalbaqali 

This looks like there was a problem when connecting to Atlas itself - I have not seen a specific error myself - can you try and run another cluster creation?

I think there was an issue with their service

Paul

**Comment by qasimalbaqali** [comment]
@stack72 Thanks for getting back to me.

I tried creating a new cluster using a new project together with new API key and I still get the same problem. Are the API key and Project ID enough to create a cluster alone? I feel like I am missing something as well.

**Comment by stack72** [comment]
So in our our acceptance tests, we are the following environment variables:

* MONGODB_ATLAS_ORG_ID
* MONGODB_ATLAS_PRIVATE_KEY
* MONGODB_ATLAS_PUBLIC_KEY

**Comment by qasimalbaqali** [comment]
Unfortunately setting these 3 environment variables also came up with the same error message.

**Comment by qasimalbaqali** [comment]
How can I help in trying to recreate this issue in order to see if this is from my end or not. I tried to create a project, add an IP whitelisting etc. with a newly created organisation and project and still I get this issue. I have assigned the following ENV variables that is mentioned in the terraform provider for their acceptance test. I am not sure why is the authentication method is done using the x509 certificate. 

```
MONGODB_ATLAS_PROJECT_ID=
MONGODB_ATLAS_ORG_ID=
MONGODB_ATLAS_PUBLIC_KEY=
MONGODB_ATLAS_PRIVATE_KEY=
```

When trying to visit `https://cloud.mongodb.com/api/atlas/v1.0/groups/5f6fa64dfa46d90c51def6fd/clusters` on the browser that is received from the error a basic auth popup is shown, and using the public and private key of the API key gets me authenticated successfully, so I doubt I am passing any wrong values.
